proposals/low-level-struct-improvements.md

```diff
@@ -939,17 +968,135 @@ This will work but is likely going to result in unnecessary code generation. One
+```
+
+This value is not related to the escape scope of the container; that is as it the container scope gets smaller it has no impact on the *ref-field-safe-to-escape* of the `ref` field values. 
+
+```c#
+ref struct Nested
+{
+    ref Span<int> Span;
+}
+
+void M(ref Nested nested)
+{
+    scoped ref Nested refLocal = nested;
+
+    // the ref-field-safe-to-escape of local is still *calling method* which means the following
+    // is illegal
+    refLocal.Span = stackalloc int[42];
+
+    scoped Nested valLocal = nested;
+
+    // the ref-field-safe-to-escape of local is still *calling method* which means the following
+    // is still illegal
+    valLocal.Span = stackalloc int[42];
+}
+```
+
+This *ref-field-safe-to-escape-scope* has essentially always existed. Up until now `ref` fields could only point to normal `struct` hence it was trivially collapsed to *calling method*.  To support `ref` fields to `ref struct` our existing rules need to be updated to take into account this new escape scope.
+
+Third the rules for ref re-assignment need to be updated to ensure that we don't violate *ref-field-safe-to-escape* for the values. Essentially for `x.e1 = ref e2` where the type of `e1` is a `ref struct` the *ref-field-safe-to-escape` must be equal. 
+
+These problems are very solvable. The compiler team has sketched out a few versions of these rules and they largely fall out from our existing analysis. The problem is there is no consuming code for such rules that helps prove out there correctness and usability. This makes us very hesitant to add support because of the fear we'll pick wrong defaults and back the runtime into usability corner when it does take advantage of this. This concern is particularly strong because .NET 8 likely pushes is in this direction with `allow T: ref struct` and `Span<Span<T>>`. The rules would be better written if it's done in conjunction with consumption code.
+
+**Decision** Delay allowing `ref` field to `ref struct` until .NET 8 where we have scenarios that will help drive the rules around these scenarios.
+
+### What will make C# 11.0?
 The features outlined in this document don't need to be implemented in a single pass. Instead they can be implemented in phases across several language releases in the following buckets:
 
 1. `ref` fields and `scoped`
-2. Sunset restricted types
-3. `unscoped` 
-4. fixed sized buffers
+2. `ref` fields to `ref struct`
+3. Sunset restricted types
+4. `unscoped` 
+5. fixed sized buffers
 
 What gets implemented in which release is merely a scoping exercise. 
 
-**Decision** Only `ref` fields, `scoped` and sunsetting restricted types will make C# 11.0. LDM is happy to revisit `unscoped` if a more natural keyword can be settled on or data suggests it's possible to make `this` `unscoped` by default in all cases.
+**Decision** Only (1) will make C# 11.0. The expectation is (2) and (3) are enabled very early in C# 12.0 to enable dogfooding by runtime throughout the .NET 8 cycle
```

Comment:
4 and 5 are then possible future?

Is there any workaround for 4? That is, could a dev manually apply `LifetimeAnnotation` with the relevant parameters and still achieve the ability to return a `ref` field on a struct or is this non-achievable until support is later added?

(Which would mean devs in need would need to continue with unsafe alternatives like `MemoryMarshal` and `Unsafe` APIs?)

Reply:
> That is, could a dev manually apply LifetimeAnnotation with the relevant parameters and still achieve the ability to return a ref field on a struct or is this non-achievable until support is later added?

Possibly. I will think on that a bit. 

The problem we're facing at the moment is: 

1. `unscoped` is really only useful for a single case. There are others but they're pretty esoteric (at least the ones we've considered). It's not really at the bar for a keyword.
2. `unscoped ref` is a good natural default for `struct` but pretty awful for `ref struct` hence can't categorically invert the default here. 

Using the attribute directly on the type / member is possibly a good middle ground. The concern I would have is that will force us to essentially find all the other uses, which are incorrect, and mark them as an error. Basically I don't want to get into a situation where devs start adding them to fields, type parameters, etc ... places we may want to use lifetime annotations in the future, and suddenly we cause friction on upgrade.